core/storage/user/gae_models.py

```diff
@@ -384,6 +384,28 @@ def get_or_create(cls, user_id):
             entity = cls(id=user_id)
         return entity
 
+    @staticmethod
+    def export_data(user_id):
+        """(Takeout) Export the user-relevant properties of UserStatsModel.
+
+        Args:
+            user_id: str. The user_id denotes which user's data to extract.
+
+        Returns:
+            dict. The user-relevant properties of UserStatsModel in a python
+                dict format.
+        """
+        user_model = UserStatsModel.get_or_create(user_id)
```

Comment:
This should not be creating, I think. I think it's OK to return None in this case.

/cc @BenHenning what do you think?

Reply:
bumping this @BenHenning @seanlip

Reply:
This should probably be a hard failure, though we can either do it here or at a higher level. It probably is clearer to do it here, though. Let's raise an exception if the user isn't found, and make it clear in the documentation that the caller is responsible for ensure the user ID passed in here corresponds to a real user.

Reply:
After following up with Sean, in this case we should probably return None since it's valid for a real user not to have data in this model.

For other models, we should decide between throwing an error and returning None based on the same thing: is it valid for the user not to have data in that model?

Reply:
addressed in latest commit